pax/plugins/peak_processing/LocalMinimumClustering.py

```diff
@@ -116,34 +116,77 @@ def split_peak(self, peak, split_points):
             yield self.build_peak(hits=hs, detector=peak.detector, left=left, right=right)
 
 
-@numba.jit(nopython=True)
 def find_split_points(w, min_height, min_ratio):
     """"Finds local minima in w,
     whose peaks to the left and right both satisfy:
       - larger than minimum + min_height
-      - larger than minimum * min_ratio
+      - larger than minimum * min_ratio # No longer required
     """
-    last_max = 0
-    min_since_max = 99999999999999.9
-    min_since_max_i = 0
-
-    for i, x in enumerate(w):
-        if x < min_since_max:
-            # New minimum since last max
-            min_since_max = x
-            min_since_max_i = i
-
-        if min(last_max, x) > max(min_since_max + min_height,
-                                  min_since_max * min_ratio):
-            # Significant local minimum: tell caller, reset both max and min finder
-            yield min_since_max_i
-            last_max = x
-            min_since_max = 99999999999999.9
-            min_since_max_i = i
-
-        if x > last_max:
-            # New max, reset minimum finder state
-            # Notice this is AFTER the split check, to accomodate very fast rising second peaks
-            last_max = x
-            min_since_max = 99999999999999.9
-            min_since_max_i = i
+    if np.max(w) < min_height:
+        return []
+
+    # Smooth sum waveform, then use first and sign of second order dirivitive
+    # to locate minima and maxima
+    left = int(np.clip(np.where(w > min_height)[0][0]-500, 0, np.inf))
+    right = int(np.clip(np.where(w > min_height)[0][-1]+500, 0, len(w)))
+
+    partial_xindex = np.arange(right-left)[::2]
+    _w = dsputils.smooth_lowess(w[left:right][::2], partial_xindex, frac=30/(right-left))
+    dw = np.diff(_w, n=1)
+    minima = np.where((np.hstack((dw, -1)) > 0) & (np.hstack((1, dw)) <= 0))[0]
+    maxima = np.where((np.hstack((dw, 1)) <= 0) & (np.hstack((-1, dw)) > 0))[0]
+
+    # We don't care about minima if it don't have maxima on both side
+    # This automatically make one less minima then maxima
+    if len(maxima) < 2:
+        return []
+    minima = minima[(minima > maxima[0]) & (minima < maxima[-1])]
+
+    # We look for the last minima between big peaks
+    good_minima = [int(partial_xindex[minima[ix]]+left) for ix in range(len(minima))
+                   if ((_w[maxima][ix+1] - _w[minima][ix] > min_height)
+                       & any(_w[maxima][:ix+1] - _w[minima][ix] > min_height))]
+
+    return good_minima
+
+
+class TailSeparation(LocalMinimumClustering):
```

Comment:
Add some description on this function

Reply:
More description are added to the function, but a relative short one for the class here.